src/compose.ts

```diff
@@ -493,7 +493,6 @@ export interface ComposeFactory<T, O extends Options> extends ComposeMixinable<T
 	/**
 	 * Extend the factory prototype with the supplied object literal, class, or factory
 	 *
-	 * @deprecated
```

Comment:
For the old `mixin` API nothing changes at all. For the new `mixin` API, here's an example of creating and mixing one in:
```typescript
const createFoo = compose({
	foo: 'original value'
}, function(instance, options?: { foo: string }) {
	instance.foo = 'initialized value';
});

const bar = compose.createMixin(createFoo)
	.extend({
		bar: 'bar'
	})
	.init((instance, options?: { foo: string, bar: string ) => {
		if (options) {
			instance.bar = options.bar;
			instance.foo = options.foo;
		}
	});

const createFooBar = createFoo.mixin(bar);

const fooBar = createFooBar({
	foo: 'final value',
	bar: 'new value'
});
```

There's several parts of that that merit more explanation:
```typescript
const createFoo = compose({
	foo: 'original value'
}, function(instance, options?: { foo: string }) {
	instance.foo = 'initialized value';
});
```
`options` is declared as an argument and typed here even though it's not used, because otherwise the generic will have to passed to the call to compose, or it will default to `{}`. That's not the end of the world but it does mean that in addition to any required properties arbitrary properties can be passed in options. 


```typescript
const bar = compose.createMixin(createFoo)
	.extend({
		bar: 'bar'
	})
	.init((instance, options?: { foo: string, bar: string ) => {
		if (options) {
			instance.bar = options.bar;
			instance.foo = options.foo;
		}
	});
```

Calling `createMixin(createFoo)` does a couple things(some of which are from the commits I just added to improve the inference a bit). It sets the `target` and `T` generics for the mixin, which basically tell it that it expects to be mixed into that type, and any initializers down the chain will have at least that type. It also looks at `createFoo` and gets the type of `options`, so that the generics don't need to be specified and it doesn't default to `{}`. 

One thing I was looking at, which I mentioned in the longer initial comment, is that we can make it so that the `Target` type is actually a compilation requirement on the mixin, in that it won't compile unless it's actually being mixed into that type. The only problem is that I can only get it to require the exact type, so it fails for subtypes.